packages/core/src/modules/proofs/protocol/v2/handlers/V2RequestPresentationHandler.ts

```diff
@@ -0,0 +1,110 @@
+import type { V2ProofService } from '..'
+import type { AgentConfig } from '../../../../../agent/AgentConfig'
+import type { Handler, HandlerInboundMessage } from '../../../../../agent/Handler'
+import type { DidCommMessageRepository } from '../../../../../storage/didcomm/DidCommMessageRepository'
+import type { MediationRecipientService } from '../../../../routing'
+import type { ProofResponseCoordinator } from '../../../ProofResponseCoordinator'
+import type { ProofRecord } from '../../../repository/ProofRecord'
+
+import { createOutboundMessage, createOutboundServiceMessage } from '../../../../../agent/helpers'
+import { ServiceDecorator } from '../../../../../decorators/service/ServiceDecorator'
+import { DidCommMessageRole } from '../../../../../storage'
+import { ProofProtocolVersion } from '../../../models/ProofProtocolVersion'
+import { V2RequestPresentationMessage } from '../messages/V2RequestPresentationMessage'
+
+export class V2RequestPresentationHandler implements Handler {
+  private proofService: V2ProofService
+  private agentConfig: AgentConfig
+  private proofResponseCoordinator: ProofResponseCoordinator
+  private mediationRecipientService: MediationRecipientService
+  private didCommMessageRepository: DidCommMessageRepository
+  public supportedMessages = [V2RequestPresentationMessage]
+
+  public constructor(
+    proofService: V2ProofService,
+    agentConfig: AgentConfig,
+    proofResponseCoordinator: ProofResponseCoordinator,
+    mediationRecipientService: MediationRecipientService,
+    didCommMessageRepository: DidCommMessageRepository
+  ) {
+    this.proofService = proofService
+    this.agentConfig = agentConfig
+    this.proofResponseCoordinator = proofResponseCoordinator
+    this.mediationRecipientService = mediationRecipientService
+    this.didCommMessageRepository = didCommMessageRepository
+  }
+
+  public async handle(messageContext: HandlerInboundMessage<V2RequestPresentationHandler>) {
+    const proofRecord = await this.proofService.processRequest(messageContext)
+    if (this.proofResponseCoordinator.shouldAutoRespondToRequest(proofRecord)) {
+      return await this.createPresentation(proofRecord, messageContext)
+    }
+  }
+
+  private async createPresentation(
+    record: ProofRecord,
```

Comment:
```suggestion
    proofRecord: ProofRecord,
```

Reply:
If it is renamed as `proofRecord` it will be a duplicate identifier as we are fetching updated `proofRecord` at line no. 68. So any other suggestion for this?